identity-iota/src/tangle/client.rs

```diff
@@ -74,23 +76,37 @@ impl Client {
   }
 
   /// Publishes an [`IotaDocument`] to the Tangle.
-  pub async fn publish_document(&self, document: &IotaDocument) -> Result<Receipt> {
-    self.publish_json(document.integration_index(), document).await
+  pub async fn publish_document(&self, document: &IotaDocument, compression: bool) -> Result<Receipt> {
+    self
+      .publish_json(document.integration_index(), document, compression)
+      .await
   }
 
   /// Publishes a [`DocumentDiff`] to the Tangle to form part of the diff chain for the integration
   /// chain message specified by the given [`MessageId`].
-  pub async fn publish_diff(&self, message_id: &MessageId, diff: &DocumentDiff) -> Result<Receipt> {
-    self.publish_json(&IotaDocument::diff_index(message_id)?, diff).await
+  pub async fn publish_diff(&self, message_id: &MessageId, diff: &DocumentDiff, compression: bool) -> Result<Receipt> {
+    self
+      .publish_json(&IotaDocument::diff_index(message_id)?, diff, compression)
+      .await
   }
 
-  /// Publishes arbitrary JSON data to the specified index on the Tangle.
-  pub async fn publish_json<T: ToJson>(&self, index: &str, data: &T) -> Result<Receipt> {
+  /// Compresses and Publishes arbitrary JSON data to the specified index on the Tangle.
+  pub async fn publish_json<T: ToJson>(&self, index: &str, data: &T, compression: bool) -> Result<Receipt> {
+    let mut compressed_data: Vec<u8>;
+    if compression {
+      compressed_data = MessageEncoding::compress_message(&data.to_json()?, MessageEncoding::JsonBrotli)?;
+      compressed_data = MessageEncoding::add_encoding_version_flag(compressed_data, MessageEncoding::JsonBrotli);
+    } else {
+      compressed_data = data.to_json_vec()?;
+      compressed_data = MessageEncoding::add_encoding_version_flag(compressed_data, MessageEncoding::Json);
+    }
+    compressed_data = MessageVersion::add_version_flag(compressed_data, MessageVersion::CURRENT);
+
     self
       .client
       .message()
       .with_index(index)
-      .with_data(data.to_json_vec()?)
+      .with_data(compressed_data)
```

Comment:
```suggestion
    let encoded_message_data = {
      if compression {
        MessageEncoding::compress_message(&data.to_json()?, MessageEncoding::JsonBrotli).map(|compressed_data| {
          MessageEncoding::add_encoding_version_flag(compressed_data, MessageEncoding::JsonBrotli)
        })?
      } else {
        data.to_json_vec().map(|uncompressed_data| {
          MessageEncoding::add_encoding_version_flag(uncompressed_data, MessageEncoding::Json)
        })?
      }
    };
    let message_data = MessageVersion::add_version_flag(encoded_message_data, MessageVersion::CURRENT);

    self
      .client
      .message()
      .with_index(index)
      .with_data(message_data)
```
is how I would have written this, but you can keep it the way it is if you prefer.

Reply:
The extra brackets in the suggestion after 

```rust
let encoded_message_data = { 
  if compression {
```
can be collapsed to 

```rust
let encoded_message_data = if compression {
```